cpp/harvest/marc_issn_lookup.cc

```diff
@@ -0,0 +1,171 @@
+    SubFieldInfo sub_field_info = SubFieldInfo(false);
+
+    while (MARC::Record record = data_source->read()) {
+        for (auto &field : record) {
+            if (field.getTag() == "022") {
+                const std::string sub_field_a(field.getFirstSubfieldWithCode('a'));
+                if ((sub_field_a != "") && (sub_field_a == issn)) {
+                    sub_field_info.Builder(record);
+                    // the issn is online version, no need to check further
+                    if (IsOnlineISSN(record))
+                        return sub_field_info;
+                } else {
+                    // maybe the issn is in subfield l
+                    const std::string sub_field_l(field.getFirstSubfieldWithCode('l'));
+                    if ((sub_field_l != "") && (sub_field_l == issn)) {
+                        sub_field_info.Builder(record);
+                        // the issn is online version, no need to check further
+                        if (IsOnlineISSN(record))
+                            return sub_field_info;
+                    }
+                }
+            }
+        }
+    }
+    return sub_field_info;
+}
+
+void UpdateSubfield(MARC::Subfields &subfields, const SubFieldInfo &sub_field_info) {
+    if (!subfields.replaceFirstSubfield('i', sub_field_info.i))
+        subfields.addSubfield('i', sub_field_info.i);
+    if (!subfields.replaceFirstSubfield('x', sub_field_info.x))
+        subfields.addSubfield('x', sub_field_info.x);
+    if (!subfields.replaceFirstSubfield('w', sub_field_info.w))
+        subfields.addSubfield('w', sub_field_info.w);
+
+    if (sub_field_info.t != "") {
+        if (!subfields.replaceFirstSubfield('t', sub_field_info.t))
+            subfields.addSubfield('t', sub_field_info.t);
+    }
+}
+
+void ISSNLookup(char **argv) {
+    auto input_file(MARC::Reader::Factory(argv[1]));
+    auto output_file(MARC::Writer::Factory(argv[3]));
+    int onprogress_counter(1);
+    std::string ppn("");
+
+    std::cout << "Updating in progress...\n";
+    while (MARC::Record record = input_file->read()) {
+        std::cout << "Record - " << onprogress_counter << "\r";
+        ppn = "";
+
+        for (auto &field : record) {
+            if (field.getTag() == "773") {
+                const std::string issn(field.getFirstSubfieldWithCode('x'));
+                if (issn != "") {
+                    SubFieldInfo sub_field_info(LookupISSNInDataSouce(argv[2], issn));
+                    if (sub_field_info.is_found) {
+                        MARC::Subfields subfields(field.getSubfields());
+
+                        // update subfield
+                        UpdateSubfield(subfields, sub_field_info);
+                        field.setSubfields(subfields);
+                    }
+                }
+            }
+        }
+        ++onprogress_counter;
+        output_file->write(record);
+    }
+}
+} // end of namespace
+
+int Main(int argc, char **argv) {
+    if (argc < 4)
```

Comment:
Since we always expect a fix amount of arguments, the check should be `if (argc != 4)`